### PR TITLE
Removed NotRegisteredException in C#

### PR DESCRIPTION
**Pull request by bernardnormier**
Also removed `FileException` and `ObjectAdapterIdInUseException`.

With this PR, various Ice remote exceptions defined in `Locator.ice` that used to be converted into the removed local exceptions (mostly `NotRegisteredException`) are simply not converted now.

This PR makes `AddAdminFacet` and `RemoveAdminFacet` more consistent the corresponds `ObjectAdapter` methods.

### CODE REVIEW

File: csharp/src/Ice/Communicator-LocatorManager.cs [hunk 1]
```diff
@@ -76,17 +76,9 @@ public void Response(LocatorInfo locatorInfo, IObjectPrx? proxy)
 
             public void Exception(LocatorInfo locatorInfo, System.Exception exc)
```
Comment (pepone) [inline csharp/src/Ice/Communicator-LocatorManager.cs]: `locatorInfo` is no longer used, can we remove this parameter?

File: csharp/src/Ice/Communicator-LocatorManager.cs [hunk 2]
```diff
@@ -76,17 +76,9 @@ public void Response(LocatorInfo locatorInfo, IObjectPrx? proxy)
...
-                catch (System.Exception ex)
-                {
-                    if (_callback != null)
-                    {
-                        _callback.SetException(ex);
-                    }
-                }
+                // Call even when _callback is null to get the logging
+                System.Exception ex = GetEndpointsException(_ref, exc);
+                _callback?.SetException(ex);
```
Comment (pepone) [inline csharp/src/Ice/Communicator-LocatorManager.cs]: Maybe we can rename this as `void TraceEndpointsException` seems this always returns the same exception, or even move the trace code here as it is not used anywhere else

```suggestion
                TraceEndpointsException(_ref, ex);
                _callback?.SetException(ex);
```

File: csharp/src/Ice/ObjectAdapter.cs [hunk 3]
```diff
@@ -1368,14 +1332,21 @@ private void UpdateLocatorRegistry(LocatorInfo? locatorInfo, IObjectPrx? proxy)
             {
                 // Ignore
             }
-            catch (System.Exception e)
+            catch (System.Exception ex)
```
Comment (pepone) [inline csharp/src/Ice/ObjectAdapter.cs]:
```suggestion
            catch (Exception ex)
```

File: csharp/src/Ice/Communicator-LocatorManager.cs (hunk 1 above)
Comment (bernardnormier) [reply csharp/src/Ice/Communicator-LocatorManager.cs]: Fixed.

File: csharp/src/Ice/Communicator-LocatorManager.cs (hunk 2 above)
Comment (bernardnormier) [reply csharp/src/Ice/Communicator-LocatorManager.cs]: Fixed as suggested.

File: csharp/src/Ice/Communicator-LocatorManager.cs [hunk 4]
```diff
@@ -74,19 +74,38 @@ public void Response(LocatorInfo locatorInfo, IObjectPrx? proxy)
...
+                        }
+                        else
+                        {
+                            s.Append($"well-known proxy = {_ref}\n");
+                        }
+                        s.Append($"reason = {ex}");
+                        communicator.Logger.Trace(communicator.TraceLevels.LocationCat, s.ToString());
                     }
                 }
+                _callback?.SetException(ex);
```
Comment (bentoi) [inline csharp/src/Ice/Communicator-LocatorManager.cs]: Shouldn't we change all the similar `if(_callback != null) { _callback.xxx() }` in this file as well?

Comment (bernardnormier) [reply csharp/src/Ice/Communicator-LocatorManager.cs]: Yes, for a future PR :).

**Pull request merged**